SalesforceCommerceCloud/pwa-kit
# chore(pwa-kit-dev): update eslint config

# Description

I didn't open an issue for this, as I wasn't sure whether to classify it as a _bug_, however this update is to include https://github.com/jsx-eslint/eslint-plugin-react/pull/3280. So that when declaring `Props` through forwardRef's generic in TypeScript components you don't get an error about missing `propTypes` definitions.



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

+ Bump `eslint-plugin-react` to latest version
+ Auto-detect React version

# How to Test-Drive This PR

Add the following component to a project using the current version of `pwa-kit-dev`:

```tsx
// index.tsx
import React from 'react';

interface Props {
  title: string
}

const MyComponent = React.forwardRef<HTMLDivElement, Props>((props, ref) => (
  <div ref={ref}>{props.title}</div> { /* <-- 🔴 Error here */ }
));
```

You'll get the following linter error:

> 'title' is missing in props validation eslint(react/prop-types)

After updating to latest version, the error is no longer present.

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
